examples/kkr_submit_scf.py

```diff
@@ -0,0 +1,108 @@
+"""
+Example calculation of fcc Cu making use of the SCF workchain for the kkrhost
+code.
+"""
+import ase.spacegroup
+from aiida import orm
+from aiida.engine import submit
+from masci_tools.io.kkr_params import kkrparams
+from aiida_kkr.workflows.kkr_scf import kkr_scf_wc
+
+
+def create_fcc_cu(a_lat=lambda: orm.Float(3.6142)) -> orm.StructureData:
+    """
+    Function to generate a fcc Cu structure
+
+    The function also makes sure that the formula and the space group number
+    are stored in the extras so that they can be more easily queried via the
+    QueryBuilder.
+
+    :param a_lat: lattice parameter for fcc Cu, defaults to lambda:orm.Float(3.6142)
+    :type a_lat: [type], optional
+    """
+    # Get the lattice distance for the fcc lattice
+    alat = 0.5*a_lat.value
+    structure = orm.StructureData(
+        cell=[[alat, alat, 0.0], [alat, 0.0, alat], [0.0, alat, alat]]
+    )
+    # Add the positions of the atoms in the unitcell
+    structure.append_atom(position=[0.0, 0.0, 0.0], symbols='Cu')
+
+    # Get the spacegroup information of the unitcell
+    sg = ase.spacegroup.get_spacegroup(structure.get_ase())
+
+    # Storing the formula as an extra so that it can easily be found
+    structure.set_extra('formula', structure.get_formula(mode='count'))
+    # Storing the spacegroup number so that it can be easily found
+    structure.set_extra('space_group_number', sg.no)
+
+    return structure
+
+
+def generate_kkr_parameters(input_parameters):
+    """
+    Function to generate a dictionary with the parameters needed for a
+    kkrhost run. It overwrites the default parameters with the provided
+    parameters
+    """
+    kkr_calculation_parameters = kkrparams(**input_parameters)
+
+    return kkr_calculation_parameters
+
+
+def main():
+    """
+    Main function to run an SCF run using the kkrhost code
+    """
+    # Define the KKR code via the code string
+    kkr_code = orm.load_code("jukkr-host@dummy")
+    # Define the Voronoi code via the code string
+    voronoi_code = orm.load_code("jukkr-voronoi@dummy")
+
+    # Generate the fcc Cu structure that will be used for the calculation
+    fcc_structure = create_fcc_cu(a_lat=orm.Float(3.6142))
+
+    # Generate a dictionary with the necessary parameters for the kkrhost
+    # calculation
+    kkr_calculation_parameters = generate_kkr_parameters(
+        input_parameters={
+            'LMAX': 2, 'RMAX': 7, 'GMAX': 65, 'NSPIN': 1, 'RCLUSTZ': 1.9
+        }
+    )
+
+    # Setting parameters for the submission of the job
+    options = {
+        'resources': {
+            # Total number of mpi processes
+            'tot_num_mpiprocs': 16,
+            # Name of the parallel environment
+            'parallel_env': 'mpi',
```

Comment:
Maybe you can put a comment here to tell that this is for a SGE scheduler?

Reply:
Sure thing, added now a check for whether the code is in an SGE or SLURM/PBS machine. And added some reasonable options